uprotocol/uattributes.proto

```diff
@@ -35,82 +35,119 @@
  option java_outer_classname = "UAttributesProto";
  option java_multiple_files = true;
 
-
-// uProtocol Header attributes that are common for all transports. <br>
-// The UAttributes message holds routing and UMessage attributes such as priority, permission, communication
-// status, etc....
+// Metadata describing a particular message's purpose, content and processing requirements.
+// Each type of message is described by a set of mandatory and optional attributes.
 message UAttributes {
-    // Required unique message identifier per
-    // https://github.com/eclipse-uprotocol/uprotocol-spec/blob/main/basics/uuid.adoc[uProtocol UUID]
+    // A unique message identifier as defined by https://github.com/eclipse-uprotocol/uprotocol-spec/blob/main/basics/uuid.adoc[uProtocol UUID].
+    // The timestamp contained in the UUID indicates the message's creation time.
     UUID id = 1;
 
-    // Message type (publish, request, response)
+    // This message's type which also indicates its purpose and determines contraints on the other properties.
     UMessageType type = 2;
 
-    // The source (address) of the message. For published events this is the producers publish topic
-    // for requests it is the calling uE respose topic (who sent the request). For responses this is the
-    // method URI from the uService. 
+    // The origin (address) of this message.
+    //
+    // Publish messages MUST have this property set to a URI representing the topic that this message is published to.
+    // Notification messages MUST have this property set to a URI representing the component that this notification originates from.
+    // RPC Request messages MUST have this property set to the URI that the service consumer expects to receive the response message at.
```

Comment:
Reading another bit you wrote on another issue, are these swapped? Wouldn't it be:
```
// RPC Request messages MUST have this property set to a URI identifying the method that has been invoked and which this message is the outcome of.
// RPC Response messages MUST have this property set to the URI that the service consumer expects to receive the response message at.
```

Reply:
Yeah, okay. It seems like the existing comment that was there does appear to be the opposite of what you have written here, right?
```
    // for requests it is the calling uE respose topic (who sent the request). For responses this is the
    // method URI from the uService. 
```
Because in my mind, the service consumer should be listening _on the sink_, not on the source.

WDYT?

Reply:
Wow, total reading comprehension fail on my part. You are 100% correct :face_exhaling: